apis/v1alpha2/shared_types.go

```diff
@@ -32,23 +32,6 @@ const (
 	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
 )
 
-const (
-	// AnnotationAppProtocol defines the protocol a Gateway should use for
-	// communication with a Kubernetes Service. This annotation must be present
-	// on the BackendPolicy resource and the protocol will apply to all Service
-	// ports that are selected by BackendPolicy.Spec.BackendRefs. If the
-	// AppProtocol field is available, this annotation should not be used. The
-	// AppProtocol field, when populated, takes precedence over this annotation.
-	// The value of this annotation must be also be a valid value for the
-	// AppProtocol field.
-	//
-	// Examples:
-	//
-	// - `gateway.networking.k8s.io/app-protocol: https`
-	// - `gateway.networking.k8s.io/app-protocol: tls`
-	AnnotationAppProtocol = "gateway.networking.k8s.io/app-protocol"
```

Comment:
This is a significant removal, but given that Kubernetes 1.18+ is so widely used already, I don't think it's worth including workarounds for older Kubernetes versions anymore. Since this one depended on BackendPolicy, it would be difficult to recreate without some kind of intermediate resource.

Reply:
I opened up https://github.com/kubernetes-sigs/gateway-api/issues/737 to track the versions we support.
I was ambivallent about the change but given where we are and where this API is, reworking this in won't have much ROI.
+1